### PR TITLE
minor simplification to logistic regression

### PR DESCRIPTION
**Pull request by benjaminsavage**
It turns out that `evaluate_per_user_attribution_circuit` was returning a vector of rows, where all rows but one were zero. This seems silly. If we are confident that the logic ensures all rows contribute zero except for one, we can just collapse each user's contribution to a single row by adding all of a user's rows together. No need for any communication, since there can never be any carries.

### CODE REVIEW

File: ipa-core/src/protocol/ipa_prf/prf_sharding/feature_label_dot_product.rs [hunk 1]
```diff
@@ -277,27 +275,27 @@ async fn evaluate_per_user_attribution_circuit<FV>(
...
     assert!(!rows_for_user.is_empty());
     if rows_for_user.len() == 1 {
-        return Ok(Vec::new());
+        return Ok(Replicated::<FV>::ZERO);
     }
     let first_row = &rows_for_user[0];
     let mut prev_row_inputs = initialize_new_device_attribution_variables(first_row);
 
-    let mut output = Vec::with_capacity(rows_for_user.len() - 1);
+    let mut output = Replicated::<FV>::ZERO;
```
Comment (akoshelev) [inline ipa-core/src/protocol/ipa_prf/prf_sharding/feature_label_dot_product.rs:290]: maybe explain why it works in a comment?

File: ipa-core/src/protocol/ipa_prf/prf_sharding/feature_label_dot_product.rs [hunk 2]
```diff
@@ -277,27 +275,27 @@ async fn evaluate_per_user_attribution_circuit<FV>(
...
-) -> Result<Vec<Replicated<FV>>, Error>
+) -> Result<Replicated<FV>, Error>
 where
     FV: SharedValue + CustomArray<Element = Boolean>,
     Replicated<FV>: BooleanArrayMul,
 {
     assert!(!rows_for_user.is_empty());
     if rows_for_user.len() == 1 {
-        return Ok(Vec::new());
+        return Ok(Replicated::<FV>::ZERO);
```
Comment (akoshelev) [inline ipa-core/src/protocol/ipa_prf/prf_sharding/feature_label_dot_product.rs:285]: it is sad that it is not covered by any test

**Review by akoshelev — APPROVED** [review]

File: ipa-core/src/protocol/ipa_prf/prf_sharding/feature_label_dot_product.rs [hunk 3]
```diff
@@ -277,30 +281,34 @@ async fn evaluate_per_user_attribution_circuit<FV>(
...
-) -> Result<Vec<Replicated<FV>>, Error>
+) -> Result<Option<Replicated<FV>>, Error>
 where
     FV: SharedValue + CustomArray<Element = Boolean>,
     Replicated<FV>: BooleanArrayMul,
 {
     assert!(!rows_for_user.is_empty());
     if rows_for_user.len() == 1 {
-        return Ok(Vec::new());
+        return Ok(None);
```
Comment (benjaminsavage) [inline ipa-core/src/protocol/ipa_prf/prf_sharding/feature_label_dot_product.rs:291]: @akoshelev - I added a user with just one row to the test in order to get code coverage for this line =).

**Comment by benjaminsavage** [comment]
I'm assuming the CI / miri (pull_request) test failure is spurious. It doesn't seem to have anything to do with these changes. @akoshelev ?

**Pull request merged**